src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -467,6 +465,172 @@ class CVulkanLogicalDevice final : public ILogicalDevice
         return !anyFailed;
     }
 
+    void updateDescriptorSets_impl(uint32_t descriptorWriteCount, const IGPUDescriptorSet::SWriteDescriptorSet* pDescriptorWrites, uint32_t descriptorCopyCount, const IGPUDescriptorSet::SCopyDescriptorSet* pDescriptorCopies)
+    {
+        constexpr uint32_t MAX_DESCRIPTOR_ARRAY_COUNT = 256u;
+
+        core::vector<VkWriteDescriptorSet> vk_writeDescriptorSets(descriptorWriteCount);
+
+        uint32_t bufferInfoOffset = 0u;
+        core::vector<VkDescriptorBufferInfo >vk_bufferInfos(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        uint32_t imageInfoOffset = 0u;
+        core::vector<VkDescriptorImageInfo> vk_imageInfos(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        uint32_t bufferViewOffset = 0u;
+        core::vector<VkBufferView> vk_bufferViews(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        core::vector<VkWriteDescriptorSetAccelerationStructureKHR> vk_writeDescriptorSetAS(descriptorWriteCount);
+
+        uint32_t accelerationStructuresOffset = 0u;
+        core::vector<VkAccelerationStructureKHR> vk_accelerationStructures(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        for (uint32_t i = 0u; i < descriptorWriteCount; ++i)
+        {
+            vk_writeDescriptorSets[i].sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+            vk_writeDescriptorSets[i].pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkWriteDescriptorSetAccelerationStructureKHR, VkWriteDescriptorSetAccelerationStructureNV, or VkWriteDescriptorSetInlineUniformBlockEXT
+
+            const IGPUDescriptorSetLayout* layout = pDescriptorWrites[i].dstSet->getLayout();
+            if (layout->getAPIType() != EAT_VULKAN)
+                continue;
```

Comment:
thats a hard error to which is necessary to log

Reply:
TODO

Reply:
Resolved.